subgraphs/friend-tech/schema.graphql

```diff
@@ -0,0 +1,806 @@
+  cumulativeUniqueBuyers: Int!
+
+  " Number of cumulative unique addresses that sold shares "
+  cumulativeUniqueSellers: Int!
+
+  " Number of cumulative unique addresses that bought or sold shares "
+  cumulativeUniqueTraders: Int!
+
+  " Number of cumulative unique addresses that registered and minted their shares on the protocol "
+  cumulativeUniqueSubjects: Int!
+
+  " Number of cumulative unique addresses that interacted with the protocol "
+  cumulativeUniqueUsers: Int!
+
+  " Total number of buy trades "
+  cumulativeBuyCount: Int!
+
+  " Total number of sell trades "
+  cumulativeSellCount: Int!
+
+  " Total number of trades "
+  cumulativeTradesCount: Int!
+
+  ##### Snapshots #####
+
+  " Daily usage metrics for this protocol "
+  usageMetrics: [UsageMetricsDailySnapshot!]! @derivedFrom(field: "protocol")
+
+  " Daily financial metrics for this protocol "
+  financialMetrics: [FinancialsDailySnapshot!]! @derivedFrom(field: "protocol")
+
+  " Day ID of the most recent daily snapshot "
+  lastSnapshotDayID: Int!
+
+  " Timestamp of the last time this entity was updated "
+  lastUpdateTimestamp: BigInt!
+}
+
+###############################
+##### Protocol Timeseries #####
+###############################
+
+type UsageMetricsDailySnapshot @entity @dailySnapshot {
+  " ID is # of days since Unix epoch time "
+  id: Bytes!
+
+  " Number of days since Unix epoch time "
+  day: Int!
+
+  " Protocol this snapshot is associated with "
+  protocol: Protocol!
+
+  " Number of unique daily active buyers "
+  dailyActiveBuyers: Int!
+
+  " Number of cumulative unique buyers "
+  cumulativeUniqueBuyers: Int!
+
+  " Number of unique daily active sellers "
+  dailyActiveSellers: Int!
+
+  " Number of cumulative unique sellers "
+  cumulativeUniqueSellers: Int!
+
+  " Number of unique daily active traders "
+  dailyActiveTraders: Int!
+
+  " Number of cumulative unique traders "
+  cumulativeUniqueTraders: Int!
+
+  " Number of unique daily active subjects "
+  dailyActiveSubjects: Int!
+
+  " Number of cumulative unique subjects "
+  cumulativeUniqueSubjects: Int!
+
+  " Number of unique daily active users "
+  dailyActiveUsers: Int!
+
+  " Number of cumulative unique users "
+  cumulativeUniqueUsers: Int!
+
+  " Number of daily buy trades "
+  dailyBuyCount: Int!
+
+  " Number of cumulative buy trades "
+  cumulativeBuyCount: Int!
+
+  " Number of daily sell trades "
+  dailySellCount: Int!
+
+  " Number of cumulative sell trades "
+  cumulativeSellCount: Int!
+
+  " Number of daily trades "
+  dailyTradesCount: Int!
+
+  " Number of cumulative trades "
+  cumulativeTradesCount: Int!
+
+  " Timestamp of when this snapshot was taken/last modified (May be taken after interval has passed) "
+  timestamp: BigInt!
+
+  " Block number of when this snapshot was taken/last modified (May be taken after interval has passed) "
+  blockNumber: BigInt!
+}
+
+type FinancialsDailySnapshot @entity @dailySnapshot {
```

Comment:
I was thinking ab immutable snapshots, but idk if there enough transactions that it will matter.

The tradeoff is that if there is not a new transaction in a neew day then the previous day snapshot is not created.